clouddriver-azure/src/test/groovy/com/netflix/spinnaker/clouddriver/azure/resources/loadbalancer/deploy/ops/DeleteAzureLoadBalancerAtomicOperationUnitSpec.groovy

```diff
@@ -0,0 +1,137 @@
+  private static final LB_RULE_BACKEND_PORT1 = 80
+  private static final LB_RULE_PROBE_NAME1 = "healthcheck1"
+  private static final LB_RULE_IDLE_TIMEOUT1 = 4
+  private static final LB_RULE_NAME2 = "lbRule2"
+  private static final LB_RULE_PROTOCOL2 = "TCP"
+  private static final LB_RULE_EXTERNAL_PORT2 = 8080
+  private static final LB_RULE_BACKEND_PORT2 = 8080
+  private static final LB_RULE_PROBE_NAME2 = "healthcheck2"
+  private static final LB_RULE_IDLE_TIMEOUT2 = 5
+
+  @Shared
+  ObjectMapper mapper = new ObjectMapper()
+
+  @Shared
+  DeleteAzureLoadBalancerAtomicOperationConverter converter
+
+  def setupSpec() {
+    TaskRepository.threadLocalTask.set(Mock(Task))
+
+    this.converter = new DeleteAzureLoadBalancerAtomicOperationConverter(objectMapper: mapper)
+    def accountCredentialsProvider = Mock(AccountCredentialsProvider)
+    def mockCredentials = Mock(AzureNamedAccountCredentials)
+    accountCredentialsProvider.getCredentials(_) >> mockCredentials
+    converter.accountCredentialsProvider = accountCredentialsProvider
+  }
+
+  void "should fail to delete a Network Load Balancer that does not exist"() {
+    setup:
+      def input = [
+        loadBalancerName: LOAD_BALANCER_NAME,
+        region: REGION,
+        accountName: ACCOUNT_NAME,
+        cloudProvider: CLOUD_PROVIDER,
+        appName: APP_NAME,
+        stack: STACK,
+        detail: DETAIL,
+        vnet: VNET,
+        securityGroups: SECURITY_GROUPS,
+        probes: [
+          [
+            probeName: PROBE_NAME1,
+            probeProtocol: PROBE_PROTOCOL1,
+            probePort: PROBE_PORT1,
+            probePath: PROBE_PATH1,
+            probeInterval: PROBE_INTERVAL1,
+            unhealthyThreshold: PROBE_UNHEALTHY_THRESHOLD1
+          ],
+          [
+            probeName: PROBE_NAME2,
+            probeProtocol: PROBE_PROTOCOL2,
+            probePort: PROBE_PORT2,
+            probeInterval: PROBE_INTERVAL2,
+            unhealthyThreshold: PROBE_UNHEALTHY_THRESHOLD2
+          ]
+        ],
+        loadBalancingRules: [
+          [
+            ruleName: LB_RULE_NAME1,
+            protocol: LB_RULE_PROTOCOL1,
+            externalPort: LB_RULE_EXTERNAL_PORT1,
+            backendPort: LB_RULE_BACKEND_PORT1,
+            probeName: LB_RULE_PROBE_NAME1,
+            idleTimeout: LB_RULE_IDLE_TIMEOUT1
+          ],
+          [
+            ruleName: LB_RULE_NAME2,
+            protocol: LB_RULE_PROTOCOL2,
+            externalPort: LB_RULE_EXTERNAL_PORT2,
+            backendPort: LB_RULE_BACKEND_PORT2,
+            probeName: LB_RULE_PROBE_NAME2,
+            idleTimeout: LB_RULE_IDLE_TIMEOUT2
+          ]
+        ]
+      ]
+
+      def description = converter.convertDescription(input)
+      //def operation = new DeleteAzureLoadBalancerAtomicOperation(description)
+
+    when:
+      def operation = converter.convertOperation(input)
+
+    then:
+      operation instanceof DeleteAzureLoadBalancerAtomicOperation
```

Comment:
Yes, it calls the specific implementation for the validator method and the actual converter.